src/Send/Context/TokenBalanceContext.tsx

```diff
@@ -0,0 +1,88 @@
+import {isEqual} from 'lodash'
+import React, {MutableRefObject, useRef} from 'react'
+import {useQuery} from 'react-query'
+
+import {useUTXOs} from '../../hooks'
+import {Logger} from '../../legacy/logging'
+import {multiTokenFromRemote, RemoteValue} from '../../legacy/utils'
+import {useSelectedWallet} from '../../SelectedWallet'
+import {
+  getDefaultNetworkTokenEntry,
+  getLatestYoroiTransaction,
+  LastYoroiTransaction,
+  MultiToken,
+  YoroiWallet,
+} from '../../yoroi-wallets'
+
+const TokenBalanceContext = React.createContext<undefined | TokenBalanceContext>(undefined)
+export const TokenBalanceProvider: React.FC = ({children}) => {
+  const wallet = useSelectedWallet()
+  const tx = getLatestYoroiTransaction(Object.values(wallet.transactions))
+  const tokenBalanceRef = useRef<TokenBalanceRef>({wallet, tx, isStale: false})
+
+  const tokenBalance = useTokenBalance(tokenBalanceRef.current.wallet)
+  useIsStale(wallet, tokenBalanceRef)
+
+  return (
+    <TokenBalanceContext.Provider
+      value={{
+        tokenBalance,
+        isStale: tokenBalanceRef.current.isStale,
+      }}
+    >
+      {children}
+    </TokenBalanceContext.Provider>
+  )
+}
+
+export const useTokenBalanceContext = () => React.useContext(TokenBalanceContext) || missingProvider()
+
+const missingProvider = () => {
+  throw new Error('TokenBalanceProvider is missing')
+}
+
+const useTokenBalance = (wallet: YoroiWallet): MultiToken => {
+  const utxos = useUTXOs(wallet)
+  const defaultMultiToken = new MultiToken([], getDefaultNetworkTokenEntry(wallet.networkId))
```

Comment:
this will create a new Multitoken on each render
this could be a performance issue